### PR TITLE
MB-9940-get-appcontext-linter-running

### PR DESCRIPTION
**Pull request by NamibiaTorres**
## Description

This PR gets our appcontext linter running against all files. This linter can not be hooked up to-precommit yet because there is a PR to fix certain files that are erroneously passing in pop.connection to structs, and this linter would interfere with that work. 

However, this running and can be tested by running it against all files and making sure that it excludes files that should call `pop.conneciton`, while throwing an error in files that shouldn't be calling pop.connection.
## Reviewer Notes

The only file that should ignore this linter is pkg/appcontext/context.go

## Setup

To test that this linter works as expected run:

```sh
go run ./cmd/appcontext-linter/main.go -- ./...
```
This runs the linter against all files with the `-- ./...` flag

You will see messages for the files that are currently being fixed in @ahobson's PR. Note: You will see a lot of files that get caught by the linter, and this is okay.

To check that the file condition is working comment out the following in `pkg/appcontext-linter/appctx.go`:
```golang
		if file.Name.Name == "appcontext" {
			return
		}
```
Then rerun the linter: `go run ./cmd/appcontext-linter/main.go -- ./...`

You should now see that the error message pops up for the following file
```
/pkg/appcontext/context.go
```

This is good because it means that our condition works when it is uncommented! This should successfully indicate that our linter runs as expected.

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-9940) for this change

### CODE REVIEW

**Comment by pearl-truss** [comment]
After I commented out the condition mentioned in the description. I seem to get a similar output as before I commented it out when re-running the linter. I'm wondering if there's another step I'm missing?

```sh
MacBook-Pro:mymove pearl$ go run ./cmd/appcontext-linter/main.go -- ./...
/Users/pearl/Projects/mymove/pkg/appcontext/context.go:22:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/db/sequence/database_sequencer.go:11:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/rateengine/rateengine.go:19:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/route/ghc_planner.go:20:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/notifications/move_approved.go:27:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/notifications/move_canceled.go:26:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/notifications/move_payment_reminder.go:28:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/notifications/move_reviewed.go:31:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/notifications/move_submitted.go:26:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/notifications/reweigh_requested.go:27:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/testingsuite/pop_suite.go:48:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/testingsuite/pop_suite_txn.go:30:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/handlers/contexts.go:76:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/services/move_task_order/move_task_order_hide_non_fake.go:19:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/services/dbtools/table_from_slice_creator.go:27:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/auth/authentication/auth.go:336:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/auth/authentication/auth.go:456:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/auth/authentication/devlocal.go:43:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/auth/authentication/devlocal.go:232:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/services/event/event.go:33:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/services/move_documents/generic_move_document_updater.go:16:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/services/move_documents/move_document_updater.go:17:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/services/move_documents/ppm_completer.go:17:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/services/move_documents/storage_expense_updater.go:19:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/services/move_documents/weight_ticket_updater.go:18:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/services/postal_codes/postal_code_validator.go:11:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/services/ppmservices/estimate_calculator.go:19:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/services/duty_stations/load_duty_stations.go:82:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/cmd/webhook-client/webhook/webhook.go:18:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/awardqueue/awardqueue.go:27:6: Please remove pop.Connection from the struct if not in appcontext
/Users/pearl/Projects/mymove/pkg/services/fuelprice/diesel_fuel_price_storer.go:66:6: Please remove pop.Connection from the struct if not in appcontext
exit status 3
```

File: pkg/appcontext-linter/appctx.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 )
 
 var AppContextAnalyzer = &analysis.Analyzer{
```
Comment (pearl-truss) [inline pkg/appcontext-linter/appctx.go:11]: Nit: I realize this wasn't part of this PR but I think this needs a comment (my linter picked it up)

Comment (NamibiaTorres) [reply pkg/appcontext-linter/appctx.go:11]: Weird, what does your linter say?

**Comment by NamibiaTorres** [comment]
> ```shell
> /Users/pearl/Projects/mymove/pkg/appcontext/context.go:22:6:
> ```

You should get a similar output after commenting out those lines. We want the linter to pick up all the files except for:
/Users/pearl/Projects/mymove/pkg/appcontext/context.go:22:6 (the top file that appears when the the condition is commented out).

File: pkg/appcontext-linter/appctx.go (hunk 1 above)
Comment (pearl-truss) [reply pkg/appcontext-linter/appctx.go:11]: nevermind, it isn't complaining anymore for some reason (earlier it was the orange line underneath for a warning)

**Comment by pearl-truss** [comment]
> > ```shell
> > /Users/pearl/Projects/mymove/pkg/appcontext/context.go:22:6:
> > ```
> 
> You should get a similar output after commenting out those lines. We want the linter to pick up all the files except for: /Users/pearl/Projects/mymove/pkg/appcontext/context.go:22:6 (the top file that appears when the the condition is commented out).

got it, thanks for clearing that up!

**Pull request merged**